File: PyNigma/keyboard.py
class Keyboard:
    """
    Keyboard and Lampboard
    """

    def __init__(self) -> None:
        self.alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

    def forward(self, letter: str) -> int:

        signal = self.alphabet.find(letter)

        return signal

    def backward(self, signal: int) -> str:

        letter = self.alphabet[signal]

        return letter

File: CNigmaPlusPlus/Keyboard.cpp
#include "pch.h"
#include "Keyboard.h"

Keyboard::Keyboard(){}

size_t Keyboard::Forward(const char& Letter)
{
    const size_t Signal = Alphabet.find(Letter);
    
    return Signal;
}

char Keyboard::Backward(const size_t& Signal)
{
    const char Letter = Alphabet[Signal];

    return Letter;
}
